### PR TITLE
Bst 2790 bootstrap file

### PR DESCRIPTION
**Pull request by Ation**

### CODE REVIEW

File: BlockSettleUILib/BSTerminalMainWindow.cpp [hunk 1]
```diff
@@ -2296,8 +2308,8 @@ void BSTerminalMainWindow::activateClient(const std::shared_ptr<BsClient> &bsCli
    connect(bsClient_.get(), &BsClient::processPbMessage, ui_->widgetChat, &ChatWidget::onProcessOtcPbMessage);
    connect(ui_->widgetChat, &ChatWidget::sendOtcPbMessage, bsClient_.get(), &BsClient::sendPbMessage);
 
-   connect(bsClient_.get(), &BsClient::ccGenAddrUpdated, this, [this](const BinaryData &ccGenAddrData) {
-      ccFileManager_->setCcAddressesSigned(ccGenAddrData);
+   connect(bsClient_.get(), &BsClient::bootstapDataUpdated, this, [this](const std::string& data) {
```
Comment (goatpig) [inline BlockSettleUILib/BSTerminalMainWindow.cpp:2311]: typo

File: BlockSettleUILib/BSTerminalMainWindow.cpp [hunk 2]
```diff
@@ -1077,7 +1079,30 @@ void BSTerminalMainWindow::initCcClient()
 void BSTerminalMainWindow::initUtxoReservationManager()
 {
    utxoReservationMgr_ = std::make_shared<bs::UTXOReservationManager>(
-      walletsMgr_, armory_, logMgr_->logger());
+            walletsMgr_, armory_, logMgr_->logger());
+}
+
+void BSTerminalMainWindow::initBootstrapDataManager()
+{
+   bootstrapDataManager_ = std::make_shared<BootstrapDataManager>(logMgr_->logger(), applicationSettings_);
```
Comment (goatpig) [inline BlockSettleUILib/BSTerminalMainWindow.cpp:1087]: what's the difference between local file vs resource? are we talking in binary embedded .rc metadata?

Comment (Ation) [reply BlockSettleUILib/BSTerminalMainWindow.cpp:1087]: Yes. Local file - default source for latest version. If file missing - either cleanup or fresh install, so we need to grab embedded from resources and save to local file.

File: BlockSettleUILib/PubKeyLoader.cpp [hunk 3]
```diff
@@ -42,100 +44,29 @@ BinaryData PubKeyLoader::loadKey(const KeyType kt) const
...
-   return [kt, bsMainWindow, appSettings] (const std::string& oldKey
+   return [kt, bsMainWindow, appSettings, bootstrapDataManager] (const std::string& oldKey
          , const std::string& newKeyHex, const std::string& srvAddrPort
          , const std::shared_ptr<FutureValue<bool>> &newKeyProm) {
-      const auto &deferredDialog = [kt, bsMainWindow, appSettings, newKeyHex, newKeyProm, srvAddrPort]{
-         QMetaObject::invokeMethod(bsMainWindow, [kt, bsMainWindow, appSettings, newKeyHex, newKeyProm, srvAddrPort] {
-            PubKeyLoader loader(appSettings);
+      const auto &deferredDialog = [kt, bsMainWindow, appSettings, bootstrapDataManager, newKeyHex, newKeyProm, srvAddrPort]{
+         QMetaObject::invokeMethod(bsMainWindow, [kt, bsMainWindow, appSettings, bootstrapDataManager, newKeyHex, newKeyProm, srvAddrPort] {
+            PubKeyLoader loader(appSettings, bootstrapDataManager);
```
Comment (goatpig) [inline BlockSettleUILib/PubKeyLoader.cpp:69]: The new key prompt GUI should warn the user the old key (local one) is signed and the new key (from remote) is not

**Review by goatpig — APPROVED** [review]

File: BlockSettleUILib/PubKeyLoader.cpp (hunk 3 above)
Comment (Ation) [reply BlockSettleUILib/PubKeyLoader.cpp:69]: Updated

**Pull request merged**